content/manager/high-availability-clusters.md

```diff
@@ -0,0 +1 @@

The cluster is dynamic, meaning that you do not need to specify the size of the cluster in advance. To ensure maximal performance, it is recommended that you do not have more than three Cloudify Managers in a cluster.

##Creating a Cluster
Create a cluster after you complete bootstrapping your Cloudify Managers. When you run the `cluster-start` command on a first Cloudify Manager, high availability is configured automatically. Use the `cluster-join` command, following bootstrapping, to add more Cloudify Managersto the cluster. The Cloudify Managers that you join to the cluster must be in an empty state, otherwise existing data will be lost. 

The data on each Cloudify Manager mirrors that of the active Cloudify Manager. Operations can only be performed on the active manager in the cluster, but are also reflected on the standby managers. Similarly, upload requests can only be sent to the active Cloudify Manager.

Within the cluster, Cloudify uses the Consul utility and internal health checks to dectect when the active Cloudify Manager is down, and which of the standbys will become active.

###Create Cluster Process
1. Complete bootstrapping a Cloudify Manager.
2. Run `cluster-start` on the clean manager to designate this Cloudify Manager instance as the active manager.
3. Run `cluster-add` on up to two other clean Cloudify Manager instances.
4. (Optional) To remove a Cloudify Manager from the cluster, run `cluster-remove-node`.

##Failure of the Master Cloudify Manager
As already mentioned, in the event that the active Cloudify Manager fails, one of the hot standbys is activated. It is important that the user investigates and fixes the issues that caused the original master to fail, or adds another Cloud Manager to the cluster, so that high availability is maintained, and to avoid having a single point of failure.

Because operations cannot be performed on a non-active manager, you will need to connect to that Cloudify Manager using the SSH protocol.

##Important Considerations
Be aware of the following considerations when you are implementing clustering.

* If the active Cloudify Manager was processing a workflow at the time it fails, the new active manager does not resume and complete the workflow.
* Policies are not synchronized between Cloudify Managers in the cluster.
* If there is a loss of connection between the Cloudify Managers in the cluster, they independently assume the `active` role. When the connection is resumed, the most recently updated Cloudify Manager becomes the active manager. Data that was accumulated on the other Cloudify Manager instances during the disconnection is not synchronized, so is lost. 
* A Cloudify Manager that is down remains in the cluster unless you remove it. To remove a Cloudify Manager, run `cluster-remove-node`.



```

Comment:
״The Cloudify Managers that you join to the cluster must be in an empty state, otherwise existing data will be lost״  - otherwise the operation will fail...
Process - item #2 does not require a clean manager, change clean to bootstrapped, at any time after bootstrap. Item 3 however requires clean managers.
Let's add a recommendation to run a cluster of 3 managers, to ensure resiliency through a failure, and lower the probability of a split brain situation.